File: leetcode/may/1209.js
var removeDuplicates = function(s, k) {
    let res = new Array(s.length);
    let duplicateCountStack = [];
    let i = 0, duplicateCount = 1;
    for (let j = 0; j < s.length; j++) {
        res[i] = s[j];
        if (s[j] === res[i-1]) {
            duplicateCount++;
        } else {
            duplicateCountStack.push(duplicateCount);
            duplicateCount = 1;
        }
        if (duplicateCount === k) {
            i -= k;
            duplicateCount = duplicateCountStack.pop();
        }
        i++;
    }
    return res.slice(0, i).join("");
};

File: leetcode/may/1209.py
class Solution:
    def removeDuplicates(self, s: str, k: int) -> str:        
        stck = []    
        
        for c in s:                            
            if stck and stck[-1][0] == c: # check if stack is not empty
                stck[-1][1]+=1
                if stck[-1][1] == k:
                    stck.pop()
            else:
                stck.append([c, 1])            
        
        return ''.join(c * cnt for c, cnt in stck)  